File: src/main/java/com/refactortalentscrm/service/InterviewService.java
package com.refactortalentscrm.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.refactortalentscrm.domain.Interview;
import com.refactortalentscrm.dto.InterviewDTO;
import com.refactortalentscrm.dto.InterviewSearchDTO;
import com.refactortalentscrm.dto.InterviewPageDTO;
import com.refactortalentscrm.dto.InterviewConvertCriteriaDTO;
import com.refactortalentscrm.service.GenericService;
import com.refactortalentscrm.dto.common.RequestDTO;
import com.refactortalentscrm.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface InterviewService extends GenericService<Interview, Integer> {

	List<Interview> findAll();

	ResultDTO addInterview(InterviewDTO interviewDTO, RequestDTO requestDTO);

	ResultDTO updateInterview(InterviewDTO interviewDTO, RequestDTO requestDTO);

    Page<Interview> getAllInterviews(Pageable pageable);

    Page<Interview> getAllInterviews(Specification<Interview> spec, Pageable pageable);

	ResponseEntity<InterviewPageDTO> getInterviews(InterviewSearchDTO interviewSearchDTO);
	
	List<InterviewDTO> convertInterviewsToInterviewDTOs(List<Interview> interviews, InterviewConvertCriteriaDTO convertCriteria);

	InterviewDTO getInterviewDTOById(Integer interviewId);







}

File: refactortalentscrm_ui/src/services/InterviewService.js
import http from "../http-common";

class InterviewService {
  getAllInterviews(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/interview/interviews`, searchDTO);
  }

  get(interviewId) {
    return this.getRequest(`/interview/${interviewId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/interview?field=${matchData}`, null);
  }

  addInterview(data) {
    return http.post("/interview/addInterview", data);
  }

  update(data) {
  	return http.post("/interview/updateInterview", data);
  }
  
  uploadImage(data,interviewId) {
  	return http.postForm("/interview/uploadImage/"+interviewId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new InterviewService();
